ast/src/main/java/com/graphicsfuzz/common/ast/decl/ArrayInfo.java

```diff
@@ -47,17 +126,18 @@ public void accept(IAstVisitor visitor) {
 
   @Override
   public ArrayInfo clone() {
-    return hasSize() ? new ArrayInfo(getSize()) : new ArrayInfo();
+    return hasSizeExpr() ? new ArrayInfo(constantSize, sizeExpr) :
```

Comment:
Even if !hasSizeExpr(), returning new ArrayInfo(constantSize, sizeExpr) will do the right thing.

I thus suggest just returning that, so that it doesn't look like the zero-argument constructor means something special here.

Reply:
Done